### PR TITLE
v0.15.0 update: Config watcher, auth service, refactoring, fixes and doc update

### PR DESCRIPTION
**Pull request by davidkornel**
Needs some work:

- [x] README Section about dev releases
- [ ] README Include a table of fields explaining them (not so sure if it's necessary)
- [x] Operator-gateway: Auth service include 

Documentation update:
- Readme now includes some of the most important features. How to use and what to know about them.

Fix and refactor(/helm):
-Changed pullPolicy in both stunner and operator helm to IfNotPresent.
(During stable release there is no point in pulling over and over again)
-Refactored stunner/values.yaml to make it more suitable for the following point.
-StartupProbe added to stunner to prevent `kubelet` from terminating the pod even before picking up the `stunnerd.conf` 

New feature: Config watcher
- Introduced config watcher as an option (default). This means an extra sidecar is added next to stunnerd.
Added a new template file for the RBAC resources.

### CODE REVIEW

**Comment by davidkornel** [comment]
latest updates:
-startup probe removed
- updated CRDs
- monitoring enabled by default
- hostnetwork option added, default false
- auth service is added next to the operator it is installed by default, but can be turned off

**Comment by levaitamas** [comment]
Resolves #11

**Comment by levaitamas** [comment]
I rebased to `main` in order to get access to the new test actions from #13

File: README.md [hunk 1]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+  limits:
+    cpu: 2
+    memory: 512Mi
+  requests:
+    cpu: 500m
+    memory: 128Mi
+```
+Meaning every started `stunner` pod will request `0.5 CPU` core and `128 mebibytes` of memory. These pods will only start in case the cluster can successfully allocate the given amount of resources. If the cluster lacks of resources it is advised to lower the requested value. This is not the only reason why the user should carefully configure the resources.
+
+About the resource limits, in order to not stress the Kubernetes scheduler it is advised to keep the limits down and scale out (increase) the number of running pods if needed. Actually, it is the way, do not be afraid to scale in Kubernetes. To learn more about the scaling process take a look at our full [scaling guide](https://github.com/l7mp/stunner/blob/main/docs/SCALING.md).
```
Comment (levaitamas) [inline README.md:85]: neat! love the xref to the fresh scaling doc :100:

File: README.md [hunk 2]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+
+### UDP multithreading
+
+Multi-threaded UDP listeners adds the ability for STUNner to run UDP listeners over multiple parallel readloops. The
+idea is to create a configurable number of UDP server sockets using `SO_REUSEPORT` and spawn a
+separate goroutine to run a parallel readloop for each. The kernel will load-balance allocations
+across the sockets/readloops per the IP 5-tuple, so the same allocation will always stay at the
+same CPU. This allows UDP listeners to scale to multiple CPUs.
```
Comment (levaitamas) [inline README.md:93]: I would mention at the end of the para that this can improve the performance

File: README.md [hunk 3]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+
+STUNner has full support for graceful shutdown in Kubernetes. This means that `stunner` pods will remain alive as long as there are active allocations via the embedded TURN server, and a pod will automatically remove itself once all allocations through it are deleted or timed out. 
+
+Note that the default TURN refresh lifetime is 10 minutes so STUNner may remain alive well after the last client goes away. This occurs when an UDP allocation is left open by a client (spontaneous UDP client-side connection closure cannot be reliably detected by the server). In such cases, after 10 mins the allocation will timeout and get deleted, which will then let `stunnerd` to terminate. 
+This feature enables the full support for graceful scale-down: the user can scale the number of `stunner` instances up and down as they wish and no harm should be made to active client connections meanwhile. 
+Caveats: 
+- currently the max lifetime for `stunner` to remain alive after deleted in 1 hour: this means that `stunner` will remain active only for 1 hour after it has been deleted/scaled-down. You can always set this in by adjusting the `terminationGracePeriod` on your `stunnerd` pods.
```
Comment (levaitamas) [inline README.md:114]: after deleted in 1 -> after being deleted is 1

File: README.md [hunk 4]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+udpMultithreading:
+  enabled: true
+  readLoopsPerUDPListener: 16
+```
+
+### Graceful shutdown
+
+STUNner has full support for graceful shutdown in Kubernetes. This means that `stunner` pods will remain alive as long as there are active allocations via the embedded TURN server, and a pod will automatically remove itself once all allocations through it are deleted or timed out. 
+
+Note that the default TURN refresh lifetime is 10 minutes so STUNner may remain alive well after the last client goes away. This occurs when an UDP allocation is left open by a client (spontaneous UDP client-side connection closure cannot be reliably detected by the server). In such cases, after 10 mins the allocation will timeout and get deleted, which will then let `stunnerd` to terminate.
```
Comment (levaitamas) [inline README.md:111]: and get deleted -> and gets deleted

File: README.md [hunk 5]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+```console
+helm repo add stunner https://l7mp.io/stunner
+helm repo update
+
+helm install prometheus stunner/stunner-prometheus
+```
+
+Learn more about how to enable, configure and use monitoring [here](https://github.com/l7mp/stunner/blob/main/doc/MONITORING.md#installation).
+
+#### Disabling monitoring
```
Comment (levaitamas) [inline README.md:165]: Disabling -> Disable

File: README.md [hunk 6]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+
+The default is to deploy STUNner with the monitoring port exposed in the STUNner pods to enable metric collection on K8s networking level.
```
Comment (levaitamas) [inline README.md:167]: The default is to deploy STUNner with ... -> By default STUNer is deployed with the monitoring port exposed in STUNner pods.

File: README.md [hunk 7]
```diff
@@ -19,11 +34,148 @@ helm install stunner-gateway-operator stunner/stunner-gateway-operator --create-
...
+
+In order to disable monitoring install the STUNner chart with the following command.
```
Comment (levaitamas) [inline README.md:169]: If you wish to avoid the port exposition, you can install STUNner with monitoring disabled:

File: README.md [hunk 8]
```diff
@@ -1,16 +1,31 @@
...
+    - [Graceful shutdown](#graceful-shutdown)
+    - [Monitoring](#monitoring)
 
 ## Important
-There are a few things to know before jumping right into using stunner
-- STUNner must installed into the same namespace as the gatewayclass and config was in case you are using STUNner with the gateway-operator
+There are a few things to know before jumping right into using STUNner
+- STUNner must installed into the same namespace as the `gatewayclass` and config was in case you are using STUNner with the gateway-operator
 
 ## Deploying
```
Comment (levaitamas) [inline README.md:19]: just _Deploy_ or _Deploy STUNner with Helm_?

**Review by levaitamas — CHANGES_REQUESTED** [review]
Thanks, this is truly great work! Made some comments, please check them.

**Review by levaitamas — APPROVED** [review]

**Pull request merged**